src/cache/encryption.rs

```diff
@@ -0,0 +1,26 @@
+#![allow(clippy::type_complexity)]
+use chacha20poly1305::{
+    consts::{B0, B1},
+    ChaChaPoly1305,
+};
+use std::sync::OnceLock;
+
+use chacha20::{
+    cipher::{
+        generic_array::GenericArray,
+        typenum::{UInt, UTerm},
+        StreamCipherCoreWrapper,
+    },
+    ChaChaCore,
+};
+
+/// OUR CIPHER, only initialised once
+pub static CIPHER: OnceLock<
+    ChaChaPoly1305<
+        StreamCipherCoreWrapper<ChaChaCore<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>>>,
```

Comment:
You might also want to specify here with a comment the reason behind this complex type